src/libraries/System.Memory/tests/ReadOnlySpan/Split.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static partial class ReadOnlySpanTests
```

Comment:
Please add the following tests for edge-cases. A few of these fail in the current implementation. See https://dotnetfiddle.net/65VtZc for some reasoning.

You can also test scenarios where white space sequences are interspersed in the buffer and/or separator.

```cs
[Fact]
public static void ValidateArguments_OverloadWithoutSeparator()
{
    ReadOnlySpan<char> buffer = default;

    SpanSplitEnumerator<char>  enumerator = buffer.Split();
    Assert.True(enumerator.MoveNext());
    Assert.Equal(new Range(0, 0), enumerator.Current);
    Assert.False(enumerator.MoveNext());

    buffer = "";
    enumerator = buffer.Split();
    Assert.True(enumerator.MoveNext());
    Assert.Equal(new Range(0, 0), enumerator.Current);
    Assert.False(enumerator.MoveNext());

    buffer = " ";
    enumerator = buffer.Split();
    Assert.True(enumerator.MoveNext());
    Assert.Equal(new Range(0, 0), enumerator.Current);
    Assert.True(enumerator.MoveNext());
    Assert.Equal(new Range(1, 1), enumerator.Current);
    Assert.False(enumerator.MoveNext());
}

[Fact]
public static void ValidateArguments_OverloadWithROSSeparator()
{
    // Default buffer
    ReadOnlySpan<char> buffer = default;

    SpanSplitEnumerator<char>  enumerator = buffer.Split(default(char));
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(0, 0));
    Assert.False(enumerator.MoveNext());

    enumerator = buffer.Split(' ');
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(0, 0));
    Assert.False(enumerator.MoveNext());

    // Empty buffer
    buffer = "";

    enumerator = buffer.Split(default(char));
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(0, 0));
    Assert.False(enumerator.MoveNext());

    enumerator = buffer.Split(' ');
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(0, 0));
    Assert.False(enumerator.MoveNext());

    // Single whitespace buffer
    buffer = " ";

    enumerator = buffer.Split(default(char));
    Assert.True(enumerator.MoveNext());
    Assert.False(enumerator.MoveNext());

    enumerator = buffer.Split(' ');
    Assert.Equal(new Range(0, 0), enumerator.Current);
    Assert.True(enumerator.MoveNext());
    Assert.Equal(new Range(0, 0), enumerator.Current);
    Assert.True(enumerator.MoveNext());
    Assert.Equal(new Range(1, 1), enumerator.Current);
    Assert.False(enumerator.MoveNext());
}

[Fact]
public static void ValidateArguments_OverloadWithStringSeparator()
{
    // Default buffer
    ReadOnlySpan<char> buffer = default;

    SpanSplitEnumerator<char> enumerator = buffer.Split(null); // null is treated as empty string
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(0, 0));
    Assert.False(enumerator.MoveNext());

    enumerator = buffer.Split("");
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(0, 0));
    Assert.False(enumerator.MoveNext());

    enumerator = buffer.Split(" ");
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(0, 0));
    Assert.False(enumerator.MoveNext());

    // Empty buffer
    buffer = "";

    enumerator = buffer.Split(null);
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(0, 0));
    Assert.False(enumerator.MoveNext());

    enumerator = buffer.Split("");
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(0, 0));
    Assert.False(enumerator.MoveNext());

    enumerator = buffer.Split(" ");
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(0, 0));
    Assert.False(enumerator.MoveNext());

    // Single whitespace buffer
    buffer = " ";

    enumerator = buffer.Split(null); // null is treated as empty string
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(0, 0));
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(1, 1));
    Assert.False(enumerator.MoveNext());

    enumerator = buffer.Split("");
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(0, 0));
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(1, 1));
    Assert.False(enumerator.MoveNext());

    enumerator = buffer.Split(" ");
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(0, 0));
    Assert.True(enumerator.MoveNext());
    Assert.Equal(enumerator.Current, new Range(1, 1));
    Assert.False(enumerator.MoveNext());
}
```